pkg/server/plugin/datastore/sql/sql.go

```diff
@@ -814,16 +804,16 @@ func (ds *sqlPlugin) ListMatchingEntries(ctx context.Context,
 	ds.mutex.Lock()
 	defer ds.mutex.Unlock()
 
-	if len(request.Selectors) < 1 {
-		return &datastore.ListSelectorEntriesResponse{}, nil
-	}
-
-	entries, err := ds.listMatchingEntries(request.Selectors)
-	if err != nil {
-		return &datastore.ListSelectorEntriesResponse{}, err
+	resp := &datastore.ListSelectorEntriesResponse{}
+	for combination := range selector.NewSetFromRaw(request.Selectors).Power() {
+		entries, err := ds.listMatchingEntries(combination.Raw())
+		if err != nil {
+			return &datastore.ListSelectorEntriesResponse{}, err
+		}
+		resp.RegisteredEntryList = append(resp.RegisteredEntryList, entries...)
 	}
 
-	resp := &datastore.ListSelectorEntriesResponse{RegisteredEntryList: entries}
+	sortRegistrationEntries(resp.RegisteredEntryList)
 	return resp, nil
```

Comment:
Should we update proto definition with a comment that the return value of this method should be sorted? If we make that assumption in consuming code, it will need to hold true for other datastore implementations too

Reply:
TBH, I was a little surprised the plugin had the sorting code. If the calling code has sorting requirements, it should probably do it there. In either case, the sorting code should probably be moved into a common package (I was going to do that in my next PR).

One thing we might consider doing is to create a validation framework that plugin authors can run against their plugins to make sure it is following whatever contract we expect. If we do maintain a requirement for the plugin to sort the entries before returning (using the provided helper for consistency), it could be asserted by the validation framework.